src/Stratis.Bitcoin.Features.PoA/PoANetwork.cs

```diff
@@ -19,10 +19,14 @@ public class PoANetwork : Network
         /// Blocks that are not signed with private keys that correspond
         /// to public keys from this list are considered to be invalid.
         /// </remarks>
-        public List<PubKey> FederationPublicKeys { get; }
+        public List<PubKey> FederationPublicKeys { get; protected set; }
+
+        public uint TargetSpacingSeconds { get; protected set; }
 
         public PoANetwork()
         {
+            this.TargetSpacingSeconds = 16;
```

Comment:
Can PoANetwork be inheriting from Network? 
TargetSpacingSeconds could be part of `PoaConsensusOptions` and FederationPublicKeys don't have to be in Network, they can be in the .config file and that makes them easier to be modified by the user (no need to rebuild).
Also PoA should be Poa everywhere (small 'a').

Reply:
>Also PoA should be Poa everywhere (small 'a').

Why? 

>Can PoANetwork be inheriting from Network?

It is inherited from network

>TargetSpacingSeconds could be part of PoaConsensusOptions and FederationPublicKeys don't have to be in Network, they can be in the .config file and that makes them easier to be modified by the user (no need to rebuild).

I think this is network specific. In config we never have values that can change consensus. 
Also not sure if having to rebuild it is a bad thing- creator of federation will have his own custom network and I expect him to override not only target spacing and fed keys but also other network values like base58 prefixes, premine amount, default ports and so on.

Reply:
> Why?

Poa like Pow or Pos or Dns or Api or Mvc :-)

> It is inherited from network

ah yes lol sorry.

> I think this is network specific. 

TargetSpacingSeconds is going to be checked in consensus rules so I see no reason why it can't be a consensus option.

For the keys, I don't think they're network specific. Network values should never change (apart from checkpoints and a few others). The pubkeys of the participants should be configurable. We already know that these are likely to be changed from time to time (members added or removed, members losing their keys or they get compromised).

> Also not sure if having to rebuild it is a bad thing

For user experience it's better if a member says to the other "my key has been compromised, here is the new one" and they can update their config file, rather than having to build, test and redistribute an executable.

> to override not only target spacing

Yeah he can do that in the PoaNetwork file, but the target spacing property is declared as part of PoaconsensusOptions.

> also other network values like base58 prefixes, premine amount, default ports and so on.

Of course

Reply:
>TargetSpacingSeconds is going to be checked in consensus rules so I see no reason why it can't be a consensus option.

Because network creator should be able to change this value. It's a network property, not consensus algho property.

>For user experience it's better if a member says to the other "my key has been compromised, here is the new one" and they can update their config file, rather than having to build, test and redistribute an executable.

Currently we don't support editing fed members except for via hardfork which also requires recompilation.

Reply:
However ok, not a bad idea on the second thought. 
I'll move those values to consensus options

Reply:
In consensus options we usually have variables that we can override or leave as it is. FedKeys are the values that we want to force people to set.